vm/vmExpr.go

```diff
@@ -734,6 +734,27 @@ func invokeExpr(expr ast.Expr, env *Env) (reflect.Value, error) {
 	case *ast.CallExpr:
 		return callExpr(e, env)
 
+	case *ast.DeleteExpr:
+		mapExpr, err := invokeExpr(e.MapExpr, env)
+		if err != nil {
+			return nilValue, newStringError(e, err.Error())
+		}
+
+		keyExpr, err := invokeExpr(e.KeyExpr, env)
+		if err != nil {
+			return nilValue, newStringError(e, err.Error())
+		}
+
+		if mapExpr.Kind() != reflect.Map {
+			return nilValue, newStringError(e, "delete only works on map")
+		}
+		if keyExpr.Kind() != reflect.String {
```

Comment:
Is there a way to remove the requirement for keyExpr to be a string? Not required, just saying it would be nice.

Reply:
Whether to create a new issue, describe this reason, and leave a todo？

Reply:
Can probably just leave it be for now. Something that can be looked into later.

Reply:
ok